src/base/settingvalue.h

```diff
@@ -40,8 +40,15 @@ class CachedSettingValue
     using ProxyFunc = std::function<T (const T&)>;
 
 public:
-    explicit CachedSettingValue(const char *keyName, const T &defaultValue = T()
-            , ProxyFunc proxyFunc = [](const T &value) { return value; })
```

Comment:
Honestly, I was sure the compiler optimizes it, and there will be no function call in default case.